### PR TITLE
create book entity and some services to save and getAll books from db…

### PR DESCRIPTION
**Pull request by don-bigdad**
… using SpringBoot framework

### CODE REVIEW

File: src/main/java/com/example/springbootbookshop/entity/Book.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (mmosii) [inline src/main/java/com/example/springbootbookshop/entity/Book.java]: better to change name of table to "books"

File: src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t save book in DB:" + book, e);
```
Comment (mmosii) [inline src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java]: maybe custom exception will be better in this case

File: src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book",Book.class)
+                    .getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can`t get all books from DB", e);
+        }
```
Comment (mmosii) [inline src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java]: maybe custom exception will be better in this case

**Review by mmosii — CHANGES_REQUESTED** [review]
Great job! Have few minor suggestions

File: .gitignore [hunk 4]
```diff
@@ -31,4 +31,5 @@ build/
 
 ### VS Code ###
 .vscode/
-src/main/resources/application.properties
\ No newline at end of file
+src/main/resources/application.properties
+src/test/resources/application.properties
```
Comment (Andriy-dz) [inline .gitignore]: why u add this?)

File: pom.xml [hunk 5]
```diff
@@ -33,16 +38,20 @@
...
-        <dependency>
-            <groupId>com.mysql</groupId>
-            <artifactId>mysql-connector-j</artifactId>
-            <scope>runtime</scope>
-        </dependency>
+            <dependency>
+                <groupId>mysql</groupId>
+                <artifactId>mysql-connector-java</artifactId>
+                <version>8.0.33</version>
+            </dependency>
```
Comment (Andriy-dz) [inline pom.xml]: set the correct number of spaces

**Review by Andriy-dz — APPROVED** [review]

**Pull request merged**